docs/cryptoswap-exchange/how-it-works.md

```diff
@@ -0,0 +1,130 @@
+
+As the market price moves, the algorithm must "rebalance" its liquidity to follow it. This action is handled carefully, because **rebalancing locks in impermanent loss**. To protect LPs, Cryptoswap only rebalances when two conditions are met:
+
+1. The internal price must move beyond a minimum threshold, called the **adjustment step**.
+2. The cost of rebalancing must be less than 50% of the trading fees earned by LPs. **This core safeguard ensures that impermanent loss is only realized when it is offset by sufficient profit.**  Ensuring LP deposits don't erode over time.
+
+!!! info "Internal EMA Price Oracle"
+      For added safety, rebalances are triggered not by the last price of the pool, but an Exponential Moving Average of all recent prices (internal price oracle), to stop manipulation of rebalances.
+      
+      In the following examples the EMA Price and Current price are assumed to be the same, however in reality the EMA will lag the current price slightly.
+
+Let's look at an example with a forex pool trading Euros (EUR) against US Dollars (USD):
+
+![Cryptoswap Liquidity & Rebalances](../assets/images/cryptoswap/cryptoswap-rebalances.png)
+
+In this scenario, the pool performs a rebalance once the price hits the adjustment step, using its collected fees to cover the cost.  
+
+### **Why Does Rebalancing Cost?**
+
+Rebalancing costs because when you are offering your assets to be swapped in return for trading fees.  As price increases, you are selling your assets.  When you rebalance, you are rebuying your assets, but at a higher price, causing a loss.  Let's look at a very simple example of a concentrated liquidity range AMM:
+
+![Rebalancing Loss](../assets/images/cryptoswap/rebalancing-loss.png)
+
+Interestly in the above example, if the user had not rebalanced, they would have kept their initial value.  However, if the price continued to increase, their loss would have been larger because they would need to buy back higher.  Rebalances need to happen, but rebalancing frequently can cause large losses, just as rebalancing rarely can, this is the balance Cryptoswap tries to automate.
+
+### **Parameters**
+
+This article from Nagaking goes into detail about each of Cryptoswap's parameters: [Deep Dive: Curve v2 Parameters](https://nagaking.substack.com/p/deep-dive-curve-v2-parameters).
+
+There are two main parameters which change change the shape of the Liquidity Curve, these are `A` and `gamma`.
+
+- **`A`**: controls liquidity concentration in the center of the bonding curve
+- **`gamma`**: controls overall breadth of the curve; can fine-tune the extremes
```

Comment:
never heard this word in english,  thats why I asked. guess is the same as  *breite* in german?

<img width="741" height="585" alt="Screenshot at 2025-08-29 13-56-39" src="https://github.com/user-attachments/assets/d03b56bc-e298-4b6f-8839-36808faf6576" />